app/controllers/application_controller.rb

```diff
@@ -450,6 +450,12 @@ def redirect_back_or_default(default, escape = true, use_escaped = true)
       begin
         uri = URI.parse(escaped_back_url)
 
+        # Remove basic auth credentials
+        uri.userinfo = nil
```

Comment:
This is not covered by a spec @oliverguenther. BTW, why do you think it necessary to remove it?